sonar-batch/src/main/java/org/sonar/batch/bootstrap/WSLoader.java

```diff
@@ -110,71 +119,94 @@ private boolean isOffline() {
     return serverStatus == NOT_ACCESSIBLE;
   }
 
-  @Nonnull
-  private byte[] loadFromCacheFirst(String id) {
-    byte[] cached = loadFromCache(id);
-    if (cached != null) {
-      return cached;
+  private void updateCache(String id, byte[] value) {
+    if (cacheEnabled) {
+      try {
+        cache.put(client.getURI(id).toString(), value);
+      } catch (IOException e) {
+        LOG.warn("Error saving to WS cache", e);
+      }
     }
+  }
 
+  @Nonnull
+  private WSLoaderResult<byte[]> loadFromCacheFirst(String id) {
     try {
-      return loadFromServer(id);
-    } catch (Exception e) {
-      if (e.getCause() instanceof HttpDownloader.HttpException) {
-        throw e;
+      return loadFromCache(id);
+    } catch (NotAvailableException cacheNotAvailable) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)